app/src/main/kotlin/com/simplemobiletools/calendar/pro/helpers/Parser.kt

```diff
@@ -60,8 +60,11 @@ class Parser {
                 } else if (repeatInterval.isXMonthlyRepetition() || repeatInterval.isXYearlyRepetition()) {
                     repeatRule = if (value.startsWith("-1")) REPEAT_ORDER_WEEKDAY_USE_LAST else REPEAT_ORDER_WEEKDAY
                 }
-            } else if (key == BYMONTHDAY && value.toInt() == -1) {
-                repeatRule = REPEAT_LAST_DAY
+            } else if (key == BYMONTHDAY) {
+                val byMonthDayValueArray = value.split(",")
+                val intFlag = byMonthDayValueArray.find { it.toInt() == -1 }
```

Comment:
instead of .find try using .any, should be easier to read. Then you wont need the intFlag variable